dashboard/app/helpers/jwt_verifier.rb

```diff
@@ -0,0 +1,58 @@
+class JwtVerifier
+  attr_reader :errors
+
+  def initialize(jwt, integration)
+    @jwt = jwt
+    @integration = integration
+    @errors = []
+  end
+
+  def verify_jwt
+    verify_audience(@jwt)
+    verify_expiration(@jwt)
+    errors.empty?
+  end
+
+  private
+
+  def verify_audience(jwt)
+    if jwt.key? :aud
+      aud = jwt[:aud]
+      aud_array = [*aud]
+      errors << 'Audience must be a string or Array of strings.' unless aud_array.all?(String)
+      if jwt.key? :azp
+        verify_azp(aud, jwt[:azp])
+      else
+        errors << 'Audience does not match client_id' unless public_key_matches_one_of_client_ids(aud)
```

Comment:
Aren't we already verifying this in the `authenticate` controller, with `integration = LtiIntegration.find_by({client_id: extracted_client_id, issuer: extracted_issuer_id})` on [line 51](https://github.com/code-dot-org/code-dot-org/pull/52476/files#diff-7992b29b7be4e17c6e1fbe1a7f569e853d780fbb4cbe826d33793a4523d1e572R57)? If the LtiIntegration table doesn't have the client_id (aud) in the jwt, we return unauthorized?

Reply:
Yes.. I kept this in here because that was how it was formatted in https://github.com/instructure/mc-lti-1.3-test-tool/blob/main/app/verifiers/jwt_verifier.rb#L30

I can actually switch it up to avoid needing it on line 51 and mostly only check here to match their structure.